pkg/resource/configmap_test.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2019 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resource
+
+import (
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	"testing"
+)
+
+func TestAddFilesToConfigMap(t *testing.T) {
+	files := map[string]string{"file1.proto": "this is the file content"}
+	emptyCm := &v1.ConfigMap{}
+	notEmptyCm := &v1.ConfigMap{Data: files}
+	type args struct {
+		files map[string]string
+		cm    *v1.ConfigMap
+	}
+	tests := []struct {
+		name string
+		args args
+	}{
+		{name: "config map has no values", args: args{
+			files: files,
+			cm:    emptyCm,
+		}},
+		{name: "config map has values", args: args{
```

Comment:
What is the purpose of this test step? If I understand it correctly then we take a CM with data, calling AddFilesToConfigMap with this CM as argument and same data as other argument, in the end checking that the data are still there.Is it so?

Reply:
This is a leftover of my TDD. I tend to test the interface first for basic functionality and then the implementation.

Reply:
Replying direct your question: yes, it's that. Because the function should override same values, and in the other use case we verify if the files are being added to the CM.

Reply:
ok, thanks